tests/scripts/mbed-build.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+
+# mbed-build.sh
+#
+# This file is part of Mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2019, Arm Limited, All Rights Reserved
+#
+# Purpose
+#
+# To run test builds of the mbed OS module for all specified targets.
+# Usage: mbed-build <folder to import to> <optinally string of space seperated targets>
+
+if [ -z "$1" ]
+then
+    echo "expected folder to import not given as parameter"
+    exit 1
+fi
+OUT_OF_SOURCE_DIR=$1
+
+if [ -z "$MBED_APP" ]
+then
+    MBED_APP='mbed-os-example-tls'
+fi
+if [ -z "$REMOTE_URL" ]
+then
+    REMOTE_URL="git@github.com:ARMmbed"
+fi
+FAILED=0
+check_tools()
+{
+    for TOOL in "$@"; do
+        if ! `hash "$TOOL" >/dev/null 2>&1`; then
+            echo "$TOOL not found!" >&2
+            exit 1
+        fi
+    done
+}
+
+mbed_build()
+{
+    PLATFORM=$1
+    COMPILER=$2
+
+    echo "*** $PLATFORM (release $COMPILER) ***"
+    mbed compile -t $COMPILER -m $PLATFORM -c
+    if [ $? -ne 0 ]
+    then
+        FAILED=1
+    fi
+
+    echo "*** $PLATFORM (debug $COMPILER) ***"
+    mbed compile -t $COMPILER -m $PLATFORM --profile mbed-os/tools/profiles/debug.json -c
+    if [ $? -ne 0 ]
+    then
+        FAILED=1
+    fi
+}
+
+create_module()
+{
+    MBED_APP_URL=$REMOTE_URL/$MBED_APP
+
+    mbed import $MBED_APP_URL
+
+    echo "'$MBED_APP_URL' imported to '$PWD'."
+}
+
+# Make sure the tools we need are available.
+check_tools "arm-none-eabi-gcc" "armcc" "mbed"
+
+cd $OUT_OF_SOURCE_DIR
+create_module
+cd $MBED_APP
+
+TOOLCHAINS="ARM GCC_ARM"
+if [ $# -eq 1 ]
+then
+TARGETS="K64F NUCLEO_F429ZI"
+else
+TARGETS=$2
```

Comment:
Please use indentation where appropriate, for example, indenting the block inside if/else blocks